src/main/java/org/jboss/logmanager/handlers/SizeRotatingFileHandler.java

```diff
@@ -235,32 +244,45 @@ protected void preWrite(final ExtLogRecord record) {
                 rotate(file);
                 // start with new file.
                 setFile(file);
-            } catch (FileNotFoundException e) {
+            } catch (IOException e) {
                 reportError("Unable to rotate log file", e, ErrorManager.OPEN_FAILURE);
             }
         }
     }
 
-    private void rotate(final File file) {
+    private void rotate(final File file) throws IOException {
         if (suffix == null) {
             // rotate.  First, drop the max file (if any), then move each file to the next higher slot.
-            new File(file.getAbsolutePath() + "." + maxBackupIndex).delete();
+            String filePath = file.getAbsolutePath();
+            File _file = new File(filePath + "." + maxBackupIndex);
+            if (_file.exists())
+                _file.delete();
```

Comment:
We can just do `new File(filePath + "." + maxBackupIndex).delete()` here. It won't fail. Either way I don't like the `_` prefix :)